### PR TITLE
Premium label

### PR DESCRIPTION
**Pull request by bjfletcher**
Add "Premium" label to cards that are premium.

image

### CODE REVIEW

File: components/title/title.js [hunk 1]
```diff
@@ -1,17 +1,23 @@
...
+			<a className="card__title-link" href={url} data-trackable="main-link">
+				<h3 className="card__title-text">
+					{title ? title.trim() : ''}
 				</h3>
 			</a>
-		);
-	}
-}
+			{isPremium ? (
+					<span className="card__title-premium">
```
Comment (ironsidevsquincy) [inline components/title/title.js]: should this be in the link?

File: components/title/title.js [hunk 2]
```diff
@@ -1,17 +1,23 @@
...
  */
-export default class extends Component {
-	render () {
-		return (
-			<a className="card__title-link" href={this.props.url} data-trackable="main-link">
-				<h3 className="card__title">
-					{this.props.title ? this.props.title.trim() : ''}
+export default ({ title, url, isPremium }) => {
+	return (
+		<div className="card__title">
```
Comment (ironsidevsquincy) [inline components/title/title.js]: Is this needed?

File: components/title/title.js (hunk 1 above)
Comment (ironsidevsquincy) [inline components/title/title.js]: also, name could be something more generic, e.g. `card__content-classification`; we might want to show other classifications

File: components/title/_title.scss [hunk 3]
```diff
@@ -32,15 +42,21 @@
...
+	margin-right: 3px;
+}
+
+.card__title-premium {
+	@include oTypographySansBold('s', $load-progressively: true);
+
+	vertical-align: bottom;
+	position: relative;
+	top: -3px;
+	padding: 2.5px 6px;
```
Comment (ironsidevsquincy) [inline components/title/_title.scss]: `2.5`?

File: components/title/_title.scss [hunk 4]
```diff
@@ -32,15 +42,21 @@
...
-		@each $breakpoint, $text-sizes in $breakpoints {
-			@include respondTo($breakpoint) {
-				@include oTypographyFallbackFontSize(serifDisplay, to-px(nth($text-sizes, 1)));
-				line-height: to-px(nth($text-sizes, 2));
-			}
-		}
-	}
+.card__title-link:after {
+	content: ' ';
+	margin-right: 3px;
```
Comment (ironsidevsquincy) [inline components/title/_title.scss]: can't this go on the `.card__title-link` directly?

File: components/title/_title.scss [hunk 5]
```diff
@@ -3,19 +3,29 @@
...
 
-.card__title {
-	@include oTypographySerifDisplay(m, $load-progressively: true);
-	color: oColorsGetPaletteColor('cold-2');
-	margin: 0;
+.card__title-text {
 	display: inline;
+	color: oColorsGetPaletteColor('cold-2');
+	font-size: inherit; // nullify H3's override
+	font-weight: 400;
```
Comment (ironsidevsquincy) [inline components/title/_title.scss]: `font-weight: inherit`?

File: components/title/title.js (hunk 1 above)
Comment (bjfletcher) [inline components/title/title.js]: I don't think tags/classifications should be part of the link as usually they go to their own links - e.g., click on "Premium" - one would expect to be taken to a page about premium (product page?)

Comment (bjfletcher) [inline components/title/title.js]: Great idea re naming.

File: components/title/_title.scss (hunk 4 above)
Comment (bjfletcher) [inline components/title/_title.scss]: If it was on the `.card__title-link` directly, then the whole title would have a right margin. That is, if the title is 3 lines, then all 3 lines would have this right margin.

What we want is for the right margin to be applied *only* to the very last line. This is a technique that I know of to achieve this.

If we moved this to the premium label as a left margin, then when the premium label is on its own on a new line, there's the unwanted left margin.

So many requirements achieved. :)

File: components/title/_title.scss (hunk 5 above)
Comment (bjfletcher) [inline components/title/_title.scss]: Ah yes, thanks. :)

File: components/title/title.js (hunk 1 above)
Comment (ironsidevsquincy) [inline components/title/title.js]: Don't think users would expect the Premium label to take you to a specific 'premium', not everything is a link that takes you somewhere, for instance the 'live' label, and it doesn't look like a link

Was just thinking it increases the hit area that someone can press to follow that link, plus would save a bit of HTML (wouldn't need the wrapped dic)

File: components/title/_title.scss (hunk 4 above)
Comment (ironsidevsquincy) [inline components/title/_title.scss]: Nice solution

File: components/title/title.js [hunk 6]
```diff
@@ -1,17 +1,23 @@
...
+			<a className="card__title-link" href={url} data-trackable="main-link">
+				<h3 className="card__title-text">
+					{title ? title.trim() : ''}
 				</h3>
 			</a>
-		);
-	}
-}
+			{isPremium ? (
+					<span className="card__title-classification">
```
Comment (ironsidevsquincy) [inline components/title/title.js]: think `card__content-classification` would be a better name - these seems to imply this is the title's classification, whereas it's the content's

File: components/title/title.js (hunk 2 above)
Comment (ironsidevsquincy) [inline components/title/title.js]: If it is, will need to rework the nested class names so they're more BEM, i.e.

 * `card__title-link` => `card__title__link`
 * `card__title-text` => `card__title__text`
 * `card__title-classification` => `card__title__content-classification`

**Pull request merged**